src/libraries/Native/Unix/System.Security.Cryptography.Native.Apple/pal_ssl.c

```diff
@@ -354,8 +354,10 @@ PAL_TlsIo AppleCryptoNative_SslRead(SSLContextRef sslContext, uint8_t* buf, uint
     return OSStatusToPAL_TlsIo(status);
 }
 
-int32_t AppleCryptoNative_SslIsHostnameMatch(SSLContextRef sslContext, CFStringRef cfHostname, CFDateRef notBefore)
+int32_t AppleCryptoNative_SslIsHostnameMatch(SSLContextRef sslContext, CFStringRef cfHostname, CFDateRef notBefore, int32_t* pOSStatus)
 {
+    *pOSStatus = noErr;
+
     if (sslContext == NULL || notBefore == NULL)
```

Comment:
```suggestion
    if (sslContext == NULL || notBefore == NULL || pOSStatus == NULL)
```

Reply:
This was resolved, but not done.

Reply:
The check is on line 359. It is different but the spirit of null check is there as far as I understood your comment.

Reply:
359 is wrong.  Since we declare the variable as `out` in the P/Invoke the shim must look exactly like

```C
if (outVar != NULL)
    *outVar = default;

if (requiredVar1 == NULL || requiredVar2 == NULL || ...)
    return -1;
```

Otherwise we're not clearing out things that we promised we'd clear.

Reply:
right now every path when outVar is not null sets value. Is that not sufficient for the 'out'

Reply:
It violates the pattern, and it doesn't hold if another out is added later.

And since you aren't supposed to assign pOSStatus in the negative return cases, once you fix the second bug you're back to the first one.

Please just follow the pattern.

Reply:
updated. Added separate diag for negative case.